src/overmap_ui.cpp

```diff
@@ -801,17 +807,67 @@ tripoint display( const tripoint &orig, const draw_data_t &data = draw_data_t()
             std::string color_notes = _( "Color codes: " );
             for( auto color_pair : get_note_color_names() ) {
                 // The color index is not translatable, but the name is.
-                color_notes += string_format( "%s:%s, ", color_pair.first.c_str(),
-                                              _( color_pair.second.c_str() ) );
+                color_notes += string_format( "%1$s:<color_%3$s>%2$s</color>, ", color_pair.first.c_str(),
+                                              _( color_pair.second.c_str() ), string_replace( color_pair.second, " ", "_" ).c_str() );
             }
-            color_notes = color_notes.replace( color_notes.end() - 2, color_notes.end(), "." );
+
+            std::string helper_text = string_format( ".\r\n \n%s\r\n%s\r\n%s\r\n ",
+                                      _( "Type GLYPH:TEXT to set a custom glyph." ),
+                                      _( "Type COLOR;TEXT to set a custom color." ),
+                                      _( "Examples: B:Base | g;Loot | !:R;Minefield" ) );
+            color_notes = color_notes.replace( color_notes.end() - 2, color_notes.end(), helper_text );
+            std::string title = _( "Note:" );
+
             const std::string old_note = overmap_buffer.note( curs );
-            const std::string new_note = string_input_popup()
-                                         .title( _( "Note (X:TEXT for custom symbol, G; for color):" ) )
-                                         .width( 45 )
-                                         .text( old_note )
-                                         .description( color_notes )
-                                         .query_string();
+            std::string new_note = old_note, tmp_note;
+
+            bool done = false, esc_pressed = false;
+            do {
+                // Popup must be created anew because the description is only
+                // printed when the window is created. This only happens once
+                // every keystroke, however.
+                string_input_popup &input_popup = string_input_popup();
```

Comment:
I guess correct way is simply `string_input_popup input_popup;`

Reply:
I've pushed a commit that addresses your review but I still get the aforementioned compiler error on my end. I'm fairly sure that everything should work, though.

Reply:
Declaring the variable _without_ calling the constructor seems to work on my end, though. I think it's because the VS compiler tries to target a deleted constructor that the other compilers simply replace with the correct one.

Reply:
Finally, KEY_ENTER maps to decimal "13", but the callback isn't called unless you register it as '\n'. So, I think I should be reverting the last change.

Reply:
As seen here: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/string_input_popup.cpp#L298-L311